solr/core/src/java/org/apache/solr/search/similarities/SchemaSimilarityFactory.java

```diff
@@ -153,6 +156,17 @@ public Similarity get(String name) {
       }
     }
 
+    private IndexSchema getSchema() {
+      // if we're using shareSchema in our config we might actually have an uninitialised
```

Comment:
I did wonder if another approach would be to only do a one-shot initialisation in `inform` i.e. if the core is already not null then basically do nothing. But I don't know enough about this code to know if genuine replacement calls can occur later and need to replace the core

Reply:
Sadly that doesn't work in `inform` as that get's called during the `SolrCore` constructor, before the index has actually been initialised.  There should probably just be some separate method to just set the schema, but yeah it's knowing how the rest of the code works that makes it tricky.  Plus I wasn't quite sure about how schema reloads would work here too.